module/support/logging/DataLogging/src/DataLogging.cpp

```diff
@@ -0,0 +1,138 @@
+
+                // Write the timestamp
+                output_file.write(reinterpret_cast<const char*>(&timestamp_us), sizeof(timestamp_us));
+
+                // Write the hash
+                output_file.write(reinterpret_cast<const char*>(&data.hash), sizeof(data.hash));
+
+                // Write the actual packet data
+                output_file.write(data.data.data(), data.data.size());
+                output_file.flush();
+
+                // Update the number of bytes we have written
+                bytes_written += size + 3 + sizeof(timestamp_us) + sizeof(data.hash);
+            });
+
+            on<Configuration, Trigger<CommandLineArguments>, Sync<DataLog>>("DataLogging.yaml")
+                .then([this](const Configuration& cfg, const CommandLineArguments& argv) {
+                    {
+                        // Get the details we need to generate a log file name
+                        config.output.directory  = cfg["output"]["directory"].as<std::string>();
+                        config.output.split_size = cfg["output"]["split_size"].as<Expression>();
+
+                        // Get the name of the currently running binary
+                        std::vector<char> data(argv[0].cbegin(), argv[0].cend());
+                        data.push_back('\0');
+                        const auto* base     = basename(data.data());
+                        config.output.binary = std::string(base);
+
+                        // Rescue any existing recorders that we want to keep
+                        std::map<std::string, ReactionHandle> new_handles;
+                        for (auto& setting : cfg["messages"].config) {
+                            std::string name = setting.first.as<std::string>();
+                            bool enabled     = setting.second.as<bool>();
+
+                            // If it was enabled and we are keeping it enabled, keep it and remove it from the old list
+                            if (handles.count(name) > 0 && enabled) {
+                                new_handles.insert(std::make_pair(name, handles[name]));
+                                handles.erase(handles.find(name));
+                            }
+                        }
+
+                        // Unbind any recorders we didn't save
+                        for (auto& handle : handles) {
+                            log<NUClear::INFO>("Data logging for type", handle.first, "disabled");
+                            handle.second.unbind();
+                        }
+
+                        // Add any recorders new ones that we don't have yet
+                        for (auto& setting : cfg["messages"].config) {
+                            std::string name = setting.first.as<std::string>();
+                            bool enabled     = setting.second.as<bool>();
+
+                            // If it was enabled and we are keeping it enabled, keep it and remove it from the old list
```

Comment:
You sure that is what this does?

Reply:
I am sure that this is not what it does? That's kinda the same...